### PR TITLE
Adjust syntax disambiguation for ref ternary.

### LINKED ISSUE #38679: Parsing issue on conditional ref operator
With the classic conditional operator, you can use an 'is' condition without any issue:
`var x = o1 is string ? o2 : o3;`
If you use the conditional ref operator `var x = o1 is string ? ref o2 : ref o3;`
 the "string ?" part is interpreted as a nullable type and the compiler emits an error on the first "ref".

### PR DESCRIPTION
**Pull request by gafter**
Fixes #38679

### CODE REVIEW

File: src/Compilers/CSharp/Test/Syntax/Parsing/ParsingTests.cs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-//#define PARSING_TESTS_DUMP
+#define PARSING_TESTS_DUMP
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Syntax/Parsing/ParsingTests.cs]: Is there a reason this was disabled before?

Comment (jcouv) [reply src/Compilers/CSharp/Test/Syntax/Parsing/ParsingTests.cs]: This should be reverted. Only turn on to troubleshoot locally I think.

---
In reply to: [324927208](https://github.com/dotnet/roslyn/pull/38713#discussion_r324927208) [](ancestors = 324927208)

File: src/Compilers/CSharp/Test/Syntax/Parsing/DeclarationExpressionTests.cs [hunk 2]
```diff
@@ -230,6 +230,69 @@ public void NullableTypeTest_04()
             EOF();
         }
 
+        [Fact]
+        public void NullableTypeTest_05()
+        {
+            UsingStatement("ref object x = o1 is string ? ref o2 : ref o3;");
```
Comment (alrz) [inline src/Compilers/CSharp/Test/Syntax/Parsing/DeclarationExpressionTests.cs]: Should `ref o1 is string ? ref o2 : ref o3` work too?

Comment (gafter) [reply src/Compilers/CSharp/Test/Syntax/Parsing/DeclarationExpressionTests.cs]: Yes, that is normally how it would be used.  I'll be sure to test that.

**Pull request merged**